### ISSUE TITLE
[Question] Need help to use this plugin 

### ISSUE THREAD

**Issue by shohan4556** [cristianbote/phaser-state-transition#16]
Hello, I have read readme file but I can not understand how to use this plugin, I added my states in Boot.js file and imported 'phaser-state-transition.min.js' file inside index.html so when I tried to start my states like this 'game.state.start('playState', [outTransition, [inTransition]]);' it shows some error in console. 
Please help me to use this plugin I know it will be a great experience.

**Comment by cristianbote** [comment]
Hey there!

Sure thing man. Let me grab some examples and I'll post them here, and add them as well as part of https://github.com/cristianbote/phaser-state-transition/issues/14.

**Comment by cristianbote** [comment]
Alright, so by default you don't need to instantiate the plugin. Just make sure you're including the dist file.
Moving on.

I notice your quote about how to transition between state, and I realized I haven't done a great job, explaining what I wanted to achieve with the new version, so sorry about that.

Take a look here https://github.com/cristianbote/phaser-state-transition/blob/master/src/index.js#L13
I'm basically overriding the default `.start` method, and make sure the state is drawn inside the system before saving. I suggest you should explore the src folder.

Bottom line:
```javascript
game.state.start('yourStateNameGoesHere', Phaser.Plugin.StateTransition.Out.SlideTop, Phaser.Plugin.StateTransition.In.SlideTop
```

If you're wondering what the hell is `Phaser.Plugin.StateTransition.Out.SlideTop` check out this file, with the default transitions: https://github.com/cristianbote/phaser-state-transition/blob/master/src/core/StateTransition.js#L43

Let me know if this was of any help.

Cheers

**Comment by shohan4556** [comment]
Thanks man it works at last, can I configure the property then how to do it.
It will be great if you add some example on you README file.

**Comment by cristianbote** [comment]
Glad that it works. Keep an eye on this repo, for more info.

**Comment by JTronLabs** [comment]
Can you explain what you mean by 

> and make sure the state is drawn inside the system before saving

When I transition to my Store state I draw all my sprites in the state's create command. But after the transition finishes they come out all wonky and misplaced.

**Comment by cristianbote** [comment]
Hey,

So, let's start with a code example, on how you're drawing inside the create func. Just post an example, and we'll see how it evolves.
It might be an issue that needs to be fixed, so I think this is the best way to be on the same page.

Looking forward!
Cheers!

**Comment by JTronLabs** [comment]
The game I am working on is [here](https://github.com/JTronLabs/Space-Shooter-Phaser/tree/0ecc6413b81e20ac6690e36d0d1e90d58fea8796). You can download the whole thing, run `npm install`, wait, `npm start` and it will open in your browser. The issue comes when you click the Play button through the store, transition successfully to the game state, beat the the level, and transition back to the store. It's at this point that the sprites get scattered.

Also included in my project is the [phaser-kinetic-scrolling-plugin](https://github.com/jdnichollsc/Phaser-Kinetic-Scrolling-Plugin), but I've removed it and problems persist.

The sprites are created in the [Store create function here](https://github.com/JTronLabs/JASIG/blob/0ecc6413b81e20ac6690e36d0d1e90d58fea8796/src/states/Store.js).

**Comment by JTronLabs** [comment]
PS: theres some major screen shake if you get damaged in-game, so be prepared. Also, the transition doesn't break 100% of the time. It might be a bit less common when kinetic scrolling is removed, but will definitely happen eventually, either way.

**Comment by JTronLabs** [comment]
It seems like game.world and game.camera's Y values are becoming way off.

**Comment by JTronLabs** [comment]
I found some issues, seems to be working now.

1) setting kinematic scroll area was wrong due to world's position changing based upon camera's position change when scrolling. In the Store state selecting a store item updates text, which changes the height of the store and calls for scrolling bounds to be updated. When this happens the camera and world can be offset from 0, and I wasn't accounting for that properly.
```
setScrollArea(){
    const top = this.currentWave.top; //top most text item
    //stateBtns position is relative to the world, which moves with camera when kineticScrolling. Account for this offset when setting new scroll bounds
    const height = this.stateBtns.y + this.stateBtns.height/2 - this.game.world.y + top; //stateBtns is a group, don't use .bottom because of issue #2

    this.game.world.setBounds(0, 0, this.game.width, height);
  }
```

2) Setting a group's top/bottom property randomly does not work (but a sprite, image, etc does). Up to this point I've used get/set on these properties and they've worked, but it seems like after adding state transitions they will randomly, occasionally break. IDK if this is an issue with the plugin, Phaser, or my code

**Comment by JTronLabs** [comment]
I started a new issue for this to try and prevent any more clutter in this issue. https://github.com/cristianbote/phaser-state-transition/issues/25
